template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/atlas/AtlasKnoxRoleConfigProvider.java

```diff
@@ -38,11 +70,67 @@ public List<String> getRoleTypes() {
         return List.of("ATLAS_SERVER");
     }
 
-    @Override
-    public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
+    private boolean isAtlasKnoxConfigurationNeeded(TemplatePreparationObject source) {
         return Objects.nonNull(source.getGatewayView())
                 && Objects.nonNull(source.getGatewayView().getExposedServices())
                 && source.getGatewayView().getExposedServices().contains(exposedServiceCollector.getAtlasService().getKnoxService());
     }
 
+    private boolean isSDXOptimalizationEnabled(TemplatePreparationObject source) {
+        return source.getStackType().equals(StackType.DATALAKE)
+                && isDatalakeVersionSupported(source.getBlueprintView().getBlueprintText())
+                && entitlementService.isSDXOptimizedConfigurationEnabled(ThreadBasedUserCrnProvider.getAccountId());
+    }
+
+    // Minimum version number is 7.2.8
+    private boolean isDatalakeVersionSupported(String blueprintText) {
```

Comment:
Add logging to this please.

Reply:
I added some, Can you check it is enough?

Reply:
A lot better, I would also add logging before the exceptions are thrown though. That way we keep track of where errors occur. Use `LOG.error` for these.

Reply:
I will go ahead and approve though since everything looks good aside from that.